docs/developing/path.md

```diff
@@ -1,45 +1,117 @@
+runs in an IFRAME, so the top-level URL and title are not automatically
+synchronized as the user navigates within your app. You can `postMessage` to the
+Sandstorm shell to ask it to update the URL in the address bar with the
+following code snippet:
+
+```js
+window.parent.postMessage({'setPath': location.pathname + location.hash}, '*');
+```
+
+In general, Sandstorm respects **deep links into grains** by copying the path
+from a grain URL into the ephemeral domain URL. For example:
+
+* Grain URL: https://sandstorm.example.com/grain/TPeYUde5rioE5keWM/foo
+
+* Ephemeral domain URL: https://96cab9109791f1254002ac1f857ecee7.sandstorm.example.com/foo
+
+The `IFRAME` also prevents the page title from propagating up into the web
+browser. You can push the current page's title into the browser's TITLE with
+the following Javascript code:
+
+```js
+window.parent.postMessage({'setTitle': document.title}, '*');
+```
+
+If you're using Meteor or another client-side routing framework, consider
+reactively watching the current route and pushing a `postMessage` event on every
+navigation. See this [reactive code
+sample](https://github.com/Azeirah/brainstorm/blob/ca01c7d2b0ae7f0480b93d7e37e19c82e37c2223/client/routes.js#L73)
+and [per-page-load code
+sample](https://github.com/paulproteus/semantic-mediawiki-sandstorm/blob/445151c033a85da5e586d1a401abea8179b599b2/resources/src/startup.js#L64).
+
+## Helping the user share access
+
+If your app wants to **create a link to itself that anyone can use**, you can
+trigger Sandstorm's "Share..." dialog with this Javascript code:
+
+```js
+window.parent.postMessage({'startSharing': {}}, '*');
+```
+
+This shares at the app's default permission level. In the future, we may extend
+this API to permit the app to choose a permission level.
+
+## Handling requests to your app's ephemeral domain
+
+The only stable URL is the grain URL, which wraps your app in the Sandstorm
+interface. However, your app might need to use its current domain name (also
+known as base URL) for:
+
+Your app might need to use base URLs for:
+
+* **Redirects**, for example after a user POSTs some data.
+
+* **Static assets** like CSS, images, Javascript.
 
 * **Links within the app**: If a user is on the home page (`/`) of an
   app, and the app wants to create a `href=` link to some other page,
   it needs to know what string to place into the `<a href>` tag.
 
-### Recommendation: Use the empty string as your base path
-
-The **easiest** way to handle Sandstorm's dynamic base path is to
-use the empty string (`''`) as your app's base path. This way,
-the app needs to make no decisions at runtime.
+### Recommendation: Use the empty string as your base URL
 
-Many web frameworks support this.
+The **easiest** way to handle Sandstorm's dynamic base URL is to use the empty
+string (`''`) as your app's base URL. This way, the app needs to make no
+decisions at runtime. Many web frameworks support this.
```

Comment:
When I first read this I was fairly sure I knew what it meant, but would have understood it clearer as "use URLs without hostnames" or even just "use relative URLs". I guess "base path" / "base url" are terms used in particular frameworks?